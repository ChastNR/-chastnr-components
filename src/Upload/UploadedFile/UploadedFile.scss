.upl__fl {
  border-radius: 8px;
  border: 1px solid #dddfe0;
  margin-top: 4px;

  .upl__fl__cntr {
    align-items: center;
    background-color: #fff;
    border-radius: inherit;
    display: flex;
    font: inherit;
    padding: 10px;
    text-decoration: none;
    word-wrap: break-word;

    &:hover {
      .upl__fl__cntr__dlt__btn {
        visibility: visible;
      }
    }

    .upl__fl__cntr__vldt {
      color: #f52222;
      display: inherit;
      margin-right: 12px;

      &.valid {
        color: #00b92d;
      }
    }

    .upl__fl__cntr__dlt__btn {
      align-self: flex-start;
      display: inherit;
      background: transparent;
      border-radius: 4px;
      border: 0;
      color: #c7c9cc;
      cursor: pointer;
      flex: 0 0 auto;
      margin-left: 12px;
      margin: 0;
      outline: none;
      padding: 0;
      visibility: hidden;

      &:hover {
        color: #9299a2;
      }
    }
  }
}
