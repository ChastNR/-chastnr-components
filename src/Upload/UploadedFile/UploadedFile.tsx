import classNames from 'clsx';
import { Close, RoundExclamation, RoundTick } from '../../base';
import { FileName } from '../FileName';
import { getFilenameAndExtension } from '../utils';

import './UploadedFile.scss';

interface IUploadedFileProps {
  file: File;
  accept?: string;
  onDeleteClick: (fileName: string) => void;
}

export const UploadedFile: React.FC<IUploadedFileProps> = ({ accept, file, onDeleteClick }) => {
  const [fileName, fileExtension] = getFilenameAndExtension(file.name);

  const isFileValid = !accept || accept.includes(fileExtension);

  const handleClick = () => onDeleteClick(file.name);

  const validationClassNames = classNames('upl__fl__cntr__vldt', {
    valid: isFileValid,
  });

  return (
    <li className="upl__fl">
      <div className="upl__fl__cntr">
        <div className={validationClassNames}>
          {isFileValid ? <RoundTick /> : <RoundExclamation />}
        </div>
        <FileName fileExtension={fileExtension} fileName={fileName} fileSize={file.size} />
        <button className="upl__fl__cntr__dlt__btn" onClick={handleClick} type="button">
          <Close />
        </button>
      </div>
    </li>
  );
};
