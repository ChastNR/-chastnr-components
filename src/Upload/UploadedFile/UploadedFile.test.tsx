import { fireEvent, render } from '@testing-library/react';
import { UploadedFile } from './UploadedFile';

describe('UploadedFile', () => {
  const testFileName = 'test.txt';
  const testFile = new File(['test'], testFileName, {
    type: 'text/plain',
  });

  const mockOnDeleteClick = jest.fn();

  it('should render properly', () => {
    const uploadedFile = render(<UploadedFile file={testFile} onDeleteClick={mockOnDeleteClick} />);

    expect(uploadedFile.container.firstChild).toMatchSnapshot();
  });

  it('should return expected results', () => {
    const uploadedFile = render(
      <UploadedFile accept=".doc" file={testFile} onDeleteClick={mockOnDeleteClick} />
    );

    const exclamationElement = uploadedFile.container.querySelector('.rnd__exl');
    const buttonElement = uploadedFile.container.getElementsByTagName('button')[0];
    fireEvent.click(buttonElement, {});

    expect(exclamationElement).toBeDefined();
    expect(mockOnDeleteClick).toBeCalledWith(testFileName);
  });
});
