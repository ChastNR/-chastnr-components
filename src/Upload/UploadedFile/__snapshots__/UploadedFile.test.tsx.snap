// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadedFile should render properly 1`] = `
<li
  class="upl__fl"
>
  <div
    class="upl__fl__cntr"
  >
    <div
      class="upl__fl__cntr__vldt valid"
    >
      <span
        class="rnd__tick"
      >
        <span>
          content
        </span>
      </span>
    </div>
    <div
      class="upl__fl__cntr__data"
    >
      <span
        class="upl__fl__cntr__name"
      >
        test
      </span>
      <span>
        .
        txt
      </span>
      <span
        class="upl__fl__cntr__size"
      >
        0.00 KB
      </span>
    </div>
    <button
      class="upl__fl__cntr__dlt__btn"
      type="button"
    >
      <span
        class="close"
      >
        <span>
          content
        </span>
      </span>
    </button>
  </div>
</li>
`;
